backend/wmg/api/v1.py

```diff
@@ -152,18 +153,49 @@ def build_ordered_cell_types_by_tissue(
 ) -> Dict[str, List[Dict[str, str]]]:
     distinct_tissues_cell_types: DataFrame = cell_counts.groupby(
         ["tissue_ontology_term_id", "cell_type_ontology_term_id"], as_index=False
-    ).first()[["tissue_ontology_term_id", "cell_type_ontology_term_id"]]
+    ).first()[["tissue_ontology_term_id", "cell_type_ontology_term_id", "n_total_cells"]]
 
-    joined = distinct_tissues_cell_types.merge(
-        cell_type_orderings, on=["tissue_ontology_term_id", "cell_type_ontology_term_id"]
+    joined = cell_type_orderings.merge(
+        distinct_tissues_cell_types, on=["tissue_ontology_term_id", "cell_type_ontology_term_id"], how="left"
     )
-    sorted = joined.sort_values(by=["tissue_ontology_term_id", "order"])
 
-    structured_result: Dict[str, List[Dict[str, str]]] = defaultdict(list)
+    # Fix depths based on the rows that need to be removed
+    joined = build_ordered_cell_types_by_tissue_fix_depths(joined)
+
+    # Remove cell types without counts
+    joined = joined[joined["n_total_cells"].notnull()]
 
-    for row in sorted.itertuples(index=False):
+    structured_result: Dict[str, List[Dict[str, str]]] = defaultdict(list)
+    for row in joined.itertuples(index=False):
         structured_result[row.tissue_ontology_term_id].append(
-            {row.cell_type_ontology_term_id: ontology_term_label(row.cell_type_ontology_term_id)}
+            {
+                "cell_type_ontology_term_id": row.cell_type_ontology_term_id,
+                "cell_type": ontology_term_label(row.cell_type_ontology_term_id),
+                "depth": row.depth,
+            }
         )
 
     return structured_result
+
+
+def build_ordered_cell_types_by_tissue_fix_depths(x):
```

Comment:
can you use a more informative param name?

Reply:
done